okhttp-android-support/src/main/java/com/squareup/okhttp/internal/huc/JavaApiConverter.java

```diff
@@ -59,15 +61,19 @@ private JavaApiConverter() {
    * Creates an OkHttp {@link Response} using the supplied {@link URI} and {@link URLConnection}
    * to supply the data. The URLConnection is assumed to already be connected.
    */
-  public static Response createOkResponse(URI uri, URLConnection urlConnection) throws IOException {
+  public static Response createOkResponseForCachePut(URI uri, URLConnection urlConnection)
+      throws IOException {
+
     HttpURLConnection httpUrlConnection = (HttpURLConnection) urlConnection;
 
     Response.Builder okResponseBuilder = new Response.Builder();
 
     // Request: Create one from the URL connection.
-    // A connected HttpURLConnection does not permit access to request headers.
-    Map<String, List<String>> requestHeaders = null;
-    Request okRequest = createOkRequest(uri, httpUrlConnection.getRequestMethod(), requestHeaders);
+
+    // Android bug 20566983: Some request headers are needed for Vary caching.
```

Comment:
I'll just kill the bug reference. I could reference issue #1589, which has the pertinent stack trace, but nobody will actually care in future that there was a bug here once.